src/Microsoft.AspNetCore.Authentication.Core/AuthenticationService.cs

```diff
@@ -0,0 +1,133 @@
+            }
+            return result;
+        }
+
+        public virtual async Task ChallengeAsync(HttpContext httpContext, string authenticationScheme, AuthenticationProperties properties, ChallengeBehavior behavior)
+        {
+            if (authenticationScheme == null)
+            {
+                var defaultChallengeScheme = await Schemes.GetDefaultChallengeSchemeAsync();
+                authenticationScheme = defaultChallengeScheme?.Name;
+                if (authenticationScheme == null)
+                {
+                    throw new InvalidOperationException($"No authenticationScheme was specified, and there was no DefaultChallengeScheme found.");
+                }
+            }
+
+            var handler = await Handlers.GetHandlerAsync(httpContext, authenticationScheme);
+            if (handler == null)
+            {
+                throw new InvalidOperationException($"No authentication handler is configured to handle the scheme: {authenticationScheme}");
+            }
+
+            var challengeContext = new ChallengeContext(httpContext, authenticationScheme, properties, behavior);
+            await handler.ChallengeAsync(challengeContext);
+        }
+
+        public virtual async Task SignInAsync(HttpContext httpContext, string authenticationScheme, ClaimsPrincipal principal, AuthenticationProperties properties)
+        {
+            if (principal == null)
+            {
+                throw new ArgumentNullException(nameof(principal));
+            }
+
+            if (authenticationScheme == null)
+            {
+                var defaultScheme = await Schemes.GetDefaultSignInSchemeAsync();
+                authenticationScheme = defaultScheme?.Name;
+                if (authenticationScheme == null)
+                {
+                    throw new InvalidOperationException($"No authenticationScheme was specified, and there was no DefaultAuthenticateScheme found.");
+                }
+            }
+
+            var handler = await Handlers.GetHandlerAsync(httpContext, authenticationScheme);
+            if (handler == null)
+            {
+                throw new InvalidOperationException($"No authentication handler is configured to handle the scheme: {authenticationScheme}");
+            }
+
+            var signInContext = new SignInContext(httpContext, authenticationScheme, principal, properties);
+            await handler.SignInAsync(signInContext);
+        }
+
+        public virtual async Task SignOutAsync(HttpContext httpContext, string authenticationScheme, AuthenticationProperties properties)
+        {
+            if (string.IsNullOrEmpty(authenticationScheme))
+            {
+                throw new ArgumentException(nameof(authenticationScheme));
+            }
+
+            var handler = await Handlers.GetHandlerAsync(httpContext, authenticationScheme);
```

Comment:
I was referring to SignOut. Should it default to SignInScheme, or should it have its own default?